src/modules/adaptors/Balancer/BalancerPoolAdaptor.sol

```diff
@@ -188,73 +239,203 @@ contract BalancerPoolAdaptor is BaseAdaptor {
-     * @param callData encoded specific txs to be used in `relayer.multicall()`. See general note at start of `Strategist Functions` section.
-     * @dev multicall() handles the actual mutation code whereas everything else mostly is there for checks preventing manipulation, etc.
-     * NOTE: possible that bpts can be in transit between AURA positions so we don't validate that the bptIn is a valid
-     *       position in the cellar during the same rebalance. Thus _liquidityGauge in params is not checked here
+     * @notice Allows strategists to exit Balancer pools using any exit.
+     * @dev The amounts in `swapsAfterExit` are overwritten by the actual amount out received from the swap.
+     * @dev `swapsAfterExit` MUST match up with expected token array returned from `_getPoolTokensWithNoPremintedBpt`.
+     *      IE if the first token in expected token array is BB A DAI, the first swap in `swapsBeforeJoin` MUST be to
+     *      swap BB A DAI.
+     * @dev Max Available logic IS NOT supported.
      */
-    function relayerExitPool(ERC20 bptIn, uint256 amountIn, ERC20[] memory tokensOut, bytes[] memory callData) public {
-        PriceRouter priceRouter = Cellar(address(this)).priceRouter();
-        uint256[] memory tokenAmount = new uint256[](tokensOut.length);
+    function exitPool(
+        ERC20 targetBpt,
+        IVault.SingleSwap[] memory swapsAfterExit,
+        SwapData memory swapData,
+        IVault.ExitPoolRequest memory request
+    ) external {
+        bytes32 poolId = IBasePool(address(targetBpt)).getPoolId();
+
+        // Figure out expected tokens out.
+        (IERC20[] memory poolTokens, , ) = vault.getPoolTokens(poolId);
+        ERC20[] memory expectedTokensOut = _getPoolTokensWithNoPremintedBpt(address(targetBpt), poolTokens);
+
+        // Insure toInternalBalance is false.
+        if (request.toInternalBalance) revert BalancerPoolAdaptor___InternalBalancesNotSupported();
+
+        // Figure out the ERC20 balance changes, and the BPT balance change from calling `exitPool`.
+        uint256[] memory tokensOutDelta = new uint256[](expectedTokensOut.length);
+        for (uint256 i; i < expectedTokensOut.length; ++i)
+            tokensOutDelta[i] = expectedTokensOut[i].balanceOf(address(this));
+
+        uint256 targetDelta = targetBpt.balanceOf(address(this));
+        vault.exitPool(poolId, address(this), payable(address(this)), request);
+        targetDelta = targetDelta - targetBpt.balanceOf(address(this));
+        for (uint256 i; i < expectedTokensOut.length; ++i)
+            tokensOutDelta[i] = expectedTokensOut[i].balanceOf(address(this)) - tokensOutDelta[i];
 
-        for (uint256 i; i < tokensOut.length; ++i) {
-            tokenAmount[i] = tokensOut[i].balanceOf(address(this));
+        PriceRouter priceRouter = Cellar(address(this)).priceRouter();
+        if (swapsAfterExit.length != expectedTokensOut.length) revert BalancerPoolAdaptor___LengthMismatch();
```

Comment:
Nit: This check could be moved to right after `expectedTokensOut` is received.

Reply:
Fixed!

Reply:
Good catch :D